spec/base_decorativeb.rb

```diff
@@ -0,0 +1,19 @@
+require 'rspec'
+require_relative '../base_decorative'
+
+RSpec.describe Decorator do
+  let(:nameable) { double('Nameable', correct_name: 'John Doe') }
+  let(:decorator) { Decorator.new(nameable) }
+
+  describe '#initialize' do
+    it 'creates a Decorator object with a nameable instance' do
+      expect(decorator.instance_variable_get(:@nameable)).to eq(nameable)
+    end
+  end
+
+  describe '#correct_name' do
+    it 'delegates the correct_name method to the nameable instance' do
+      expect(decorator.correct_name).to eq('John Doe')
+    end
+  end
```

Comment:
- [ ] Please fix the errors in this class test as shown below in the screenshot. 🙏 
![base_decorator](https://github.com/faisalakb/Ruby_OOP_school_library/assets/47625407/201c910d-97ad-4118-a013-89ccab5d9f0b)